### ISSUE TITLE
Failing test-suite

### ISSUE THREAD

**Issue by nico202** [aconchillo/guile-json#67]
Hi, I was testing guile-json with some of the tests described [here](http://seriot.ch/parsing_json.php).

```json
[,1]
```
This invalid json is accepted by guile-json (parsed as: '(1) )  while it should fail.

Also I get 2 segmentation fault on guile 3.0.4 on some files in the testsuite

Files causing a segfault:
```json
[0.4e00669999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999969999999006]
```
```json
[123e-10000000]
```

**Comment by nico202** [comment]
I run all the test suite (with the attached code), failing tests (success when they should fail):
```
SUCCESS Error in file n_string_unescaped_tab.json
SUCCESS Error in file n_string_unescaped_newline.json
SUCCESS Error in file n_string_unescaped_ctrl_char.json
SUCCESS Error in file n_number_with_leading_zero.json
SUCCESS Error in file n_number_neg_int_starting_with_zero.json
SUCCESS Error in file n_number_1_000.json
SUCCESS Error in file n_number_-01.json
SUCCESS Error in file n_array_missing_value.json
SUCCESS Error in file n_array_inner_array_no_comma.json
SUCCESS Error in file n_array_comma_and_number.json
SUCCESS Error in file n_array_1_true_without_comma.json
```

```guile
(use-modules (json)
	     (ice-9 ftw)
	     (ice-9 match)
	     (srfi srfi-13))

(define remove-stat
  ;; Remove the `stat' object the `file-system-tree' provides
  ;; for each file in the tree.
  (match-lambda
    ((name stat)              ; flat file
     name)
    ((name stat children ...) ; directory
     (list name (map remove-stat children)))))

(define list-dir
  (λ (path-to-dir)
    (remove-stat (file-system-tree path-to-dir))))

(let ((dir "/home/nixo/git/JSONTestSuite/test_parsing"))
  (map
   (lambda (file)
     (let* ((port (open-file (string-append dir "/" file) "r")))
       (with-exception-handler
	   (lambda (x)
	     (when (not (or (string-prefix? "i" file)
			    (string-prefix? "n" file)))
	       (display (format #f "FAIL Error in file ~a\n" file)))
	     (close port))
	 (lambda ()
	   (json->scm port)
	   (close port)
	   (when (not (or (string-prefix? "i" file)
			  (string-prefix? "y" file)))
	     (display (format #f "SUCCESS Error in file ~a\n" file))))
	 #:unwind? #t)))
   (cadr (list-dir dir))))
```

**Comment by aconchillo** [comment]
Hi @nico202. Thank you for reporting this. I was not aware of that test suite.

**Comment by aconchillo** [comment]
Should be fixed in 4.4.1. Thanks again!
